tests/forms_tests/tests/test_forms.py

```diff
@@ -3671,6 +3671,28 @@ class DataForm(Form):
         self.assertTrue(f.is_valid())
         self.assertEqual(f.cleaned_data, {'data': 'xyzzy'})
 
+    def test_data_multi_value_dict(self):
+        p = Person()
+
+        self.assertEqual(p.data.getlist(None), [])
+
+        p.data.setlistdefault('first_name')
+        self.assertEqual(p.data.getlist('first_name'), [])
+
+        p.data.setlist('first_name', ['John', 'Winston'])
+        self.assertEqual(p.data.getlist('first_name'), ['John', 'Winston'])
+
+    def test_files_multi_value_dict(self):
+        p = Person()
+
+        self.assertEqual(p.files.getlist(None), [])
+
+        p.files.setlistdefault('file1')
+        self.assertEqual(p.files.getlist('file1'), [])
+
+        p.files.setlist('file1', ['resume.txt', 'resume1.txt'])
+        self.assertEqual(p.files.getlist('file1'), ['resume.txt', 'resume1.txt'])
+
```

Comment:
I'm not sure we need to test implementation details of `MultiValueDict`. It seems like the following test should be enough

```python
def test_empty_data_files_multivalue_dict(self):
    form = Person()
    self.assertIsInstance(form.data, MultiValueDict)
    self.assertIsInstance(form.files, MultiValueDict)

Reply:
Yes, you are right! I've got too deep into implementation details. I've updated accordingly.